cli/cli_app.go

```diff
@@ -61,6 +62,14 @@ func CreateTerragruntCli(version string) *cli.App {
    Moreover, for the apply and destroy commands, Terragrunt will first try to acquire a lock using DynamoDB. For
    documentation, see https://github.com/gruntwork-io/terragrunt/.`
 
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "config",
+			Value: os.Getenv("TERRAGRUNT_CONFIG"),
```

Comment:
The code snippet I have above will find the `.terragrunt` file as follows:

1. A path explicitly specified via the `--terragrunt-config` option.
1. A path specified via the `TERRAGRUNT_CONFIG` environment variable.
1. Finally, fallback to the default of `.terragrunt` in the current directory.

Doesn't that support your use case?